### PR TITLE
feat(layout): RFC-060 — HorizontalStack as a default-on scored candidate

### PR DESCRIPTION
**Pull request by storkme**
> **Renumbered RFC-059 → RFC-060**: #509 claimed 059 on main first (`rfc-059-di-coupling-assignment.md`); the registry's chronological rule applies. All K59-* kill criteria are now K60-*. Branch has merged `origin/main` (through `0f0f7a0`) and the merged tree is fully green: **1045 passed / 0 failed** with the re-refreshed zone-cache pin.

## Intent

Implements [RFC-060](../blob/claude/ui-simplification-w8y9p6/docs/rfc-060-horizontal-stack-candidate.md), the graduation proposed in #513: the horizontal-stack row layout (RFC-013) becomes a default-on scored candidate in the decomposition search, mirroring DI's lifecycle — so the engine picks it per-layout where it strictly improves, and the sidebar's Row layout knob can eventually retire (#512). Evidence base: the `full_knob_sweep` winner map (horizontal wins or ties all 8 contested corpus cases; the shipped default never won one and hard-refused `ec@15 am3 plates`).

**K60-3 cleared (2026-07-31, harness-capable machine): no flipped case sims below native.** Delivered rates, candidate vs candidate-off, long warmup (216k ticks; pu3 288k):

| case | candidate (shipped default) | candidate-off | note |
|---|---|---|---|
| `ac@5` | 3.75/s (−25%) | **0.00/s** | native E4/W11 deadlocks outright |
| `ac@7` | 6.00/s (−14.3%) | **0.00/s** | native E10/W64 deadlocks |
| `pu@3` | 2.53/s (−15.7%, converged; flat vs warmup 288k→432k) | **0.00/s** | native E11/W149 deadlocks; first run pair invalidated by a harness defect, re-run on the fixed kit |
| `ec@15` | 14.45/s (−3.6% WARN) | 14.18/s (−5.5% WARN) | off arm = cell-composed rescue; raw bus still refuses |

(`pu@2` dropped out of the flipped set with the density arm — bit-identical arms, nothing to compare.) The candidate's absolute deficits on `ac@5`/`ac@7`/`pu@3` are the **flux blind spot, now measured and mechanism-identified**: sideload-fed cable taps deliver one lane (~7.5/s) where the lane walker credits two — tracked as #519 with tile-level forensics. Comparative gate (K60-3's letter and spirit): holds everywhere; the winner key is not lying.

## Scope

- **In:** `HorizontalStackCandidate` + `horizontal_choice` (strictly-better-only, ties→native), the DI-vs-horizontal pairwise resolution, `ranking_len` confinement extended to index 6, warnings-first ordering in the #392 error-free refusal tier, `any_dual_input_row` eligibility short-circuit, forced-DI stand-down, candidate-independence pins in cell geometry derivation (extract + 2× mega), `LayoutOptions.horizontal_candidate` (wasm pins `true`; no web-surface change), pinning test + fixture adjudications + `full_knob_sweep` (pure baselines + shipped-default column), zone-cache pin refresh (pre- and post-merge), RFC-060 + registry row, status.md tier-4 ladder update, merge from main (conflicts: registry numbering, wasm `?compact=1` param — both changes kept). **Added during K60-3 (2026-07-31):** sim-harness fluid-feed isolation fix (adjacent fluid ports' infinity-pipes merged into one network — pu3's water trunk carried crude; staggered ug-pipe runs now isolate each feed) + `rfc060_sim_export`, a tracked `#[ignore]` e2e exporter closing the RFC-050 "gitignored manifest generator" reproducibility gap.
- **Out:** the density arm (`equal_issues_and_denser`) — dropped in v1 after measuring its blast radius (10 flipped clean pinned artifacts for ≤5% entity shaves; decision log); web UI changes (belongs to #512); Strategy/P2 re-evaluation (needs a wider corpus per #513); DI-inside-horizontal hybrid passes; fixing the #519 lane-flux validator gap (pre-existing class, tracked).

## Verification

- [x] `cargo test --manifest-path crates/core/Cargo.toml` — **1045 passed / 0 failed on the merged tree** (`--no-fail-fast`, CI zone-cache pin)
- [x] `cargo clippy` clean on core + wasm-bindings (no new warnings)
- [x] `full_knob_sweep` re-run: shipped-default column matches design per case — strict wins fire (`ac@5` E4/W11→0/0, `ac@7` E10/W64→0/0, `pu@3` E11/W149→0/W35), `ec@15` refusal now builds (candidate=horizontal-stack), every clean layout stays bit-identical native (`ec@20` 1392, `pu@2` 6499), ties stay native (`ec@10` W1), DI keeps `steel@5`
- [x] Pinning test `horizontal_candidate_never_degrades_a_succeeding_bus_layout` (component-wise floors + bit-identity on a no-dual-input chain)
- [x] K60-1 runtime: e2e 129.7s → 190.2s = **1.47×**, inside the 1.5× budget
- [x] **Sim (K60-3) — cleared.** All four flipped cases, both arms, long warmup, real headless 2.0.77: table above; full per-run numbers, window series, and machine censuses in the RFC decision log. Harness defect found/fixed mid-verification (fluid-feed cross-contamination; unit tests added); invalid first pu3 runs discarded, not cited.
- [x] WASM rebuild + browser eyeball of `ac@5` and `ec@15` (verification protocol step 2): both render connected — `ec@15` visibly the 28×27 horizontal-stack geometry, output belt reaching the east edge; no orphaned belts on either.
- [ ] Snapshot inspection — not applicable to a specific bug; the sweep + suite are the behavioral evidence

## Deviations from intent

- **Density arm dropped from v1** (#513 proposed mirroring DI's full rule). Measured mid-flight: it flipped ten clean pinned artifacts for ≤5% entity shaves while contributing none of the headline value. Recorded in the RFC decision log; re-enable candidate alongside RFC-058.
- **#392 error-free tier reordered** (warnings before score) — not in the original proposal; forced by horizontal being the third refusal-resolver (DI's clean `ec@15` must not lose to a denser 6-warning layout). Pinned by `cell_candidate_resolves_ec15_refusal`.
- **`ec@15` default winner changed DI→horizontal** (equal 0/0, 252 vs 272 entities); the fixture now pins the succession contract (error-free, beats composition, never regresses past the 272-entity floor) instead of the champion's internal structure.
- **RFC renumbered 059→060 mid-flight** (registry collision with #509; chronological rule applied).
- **Sim-harness fix rode along** (2026-07-31): K60-3 could not produce a valid pu3 measurement without it, so the fluid-feed isolation fix landed on this branch rather than as a separate PR; decision log records the forensics. No blessed baseline exercises fluid feeds, so no re-bless.
- **`ac@5`/`ac@7` sim FAILs recorded, gate still cleared**: K60-3 is comparative (candidate vs native) and native delivers zero. The absolute deficits are a validator calibration gap (#519), not a candidate regression — the candidate strictly improved every measured case.

## Models / contracts touched

- `docs/rfc-060-horizontal-stack-candidate.md` (new, Active) + registry row; decision log carries every mid-flight call above.
- `docs/status.md` tier-4 ladder: advanced-circuit from-ore AM2 now fully clean (the long-standing `input-rate-delivery` residual is deleted by the candidate).
- Decomposition-search selection contract: two new rules recorded — forced modes beat candidates, and geometry-derivation paths are candidate-independent by construction.
- Sim-harness boundary-kit contract: fluid feeds are now isolation-guaranteed (per-slot staggered ug-runs); rates remain UNCALIBRATED per RFC-050.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01YV4Ah3pCPYkjHxQLGafUoc
https://claude.ai/code/session_016dQiVi8Uh9wEiyA9xR4uk8

### CODE REVIEW

File: docs/status.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ below are gated by the default (non-ignored) suite unless noted.
 | 1 | `iron-gear-wheel` | 1 recipe, 1 solid input | SOLVED — clean, incl. 20/s |
 | 2 | `electronic-circuit` | 2 recipes, 2 solid inputs | SOLVED — clean from ores; stress-gated at 20/22/23/30/35/40/s (yellow) and 60/s (red) from ore |
 | 3 | `plastic-bar` | 1 recipe, 1 fluid + 1 solid input | SOLVED — clean, incl. from crude; sulfuric-acid, heavy-oil cracking, and multi-machine advanced-oil-processing also gated at this tier |
-| 4 | `advanced-circuit` | 5+ recipes, mixed solid/fluid | SOLVED — from plates fully clean; from ore (AM2) green with 1 known input-rate-delivery warning (pre-existing demand-pull modeling residual). Partitioned 4/s + 5/s and horizontal-stack 7/s stress-gated. |
+| 4 | `advanced-circuit` | 5+ recipes, mixed solid/fluid | SOLVED — from plates fully clean; from ore (AM2) **fully clean since RFC-060** (the horizontal-stack candidate wins strictly-better and deletes the long-standing input-rate-delivery residual). Partitioned 4/s + 5/s and horizontal-stack 7/s stress-gated. |
```
[the remark this answers is not in the document]
Comment (storkme) [reply docs/status.md]: Accepted — applied your suggested wording in 4e8304ad ("validator-clean … but **not sim-verified**", with the live #519 link). Fair catch: the row was written from the warning-count drop before this same PR's sim evidence showed what that's worth for this topology class.

File: docs/rfc-060-horizontal-stack-candidate.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  delivery drops below native's by >5% of plan → the static winner key
+  is lying for that class; revert default pending a fixed comparator.
+  **This criterion gates the merge**, not the implementation: the PR
+  stays draft until the flipped cases (`ac@5`, `ac@7`, `pu@2`, `pu@3`,
+  `ec@15`) sim at/above native.
+- **K60-4 (scope)**: net engine LOC (excluding tests/docs) exceeds ~400
+  → the integration point is wrong; stop and redesign rather than
+  ballooning.
+
+## Verification plan
```
[the remark this answers is not in the document]
Comment (storkme) [reply docs/rfc-060-horizontal-stack-candidate.md:93]: Accepted — K60-4 dispositioned in the decision log (4e8304ad): net engine LOC on the final merged tree is **+244 −25 = 219** (`crates/core/src` + `crates/wasm-bindings/src`, tests/docs excluded), under the ~400 bar. The entry also states K60-2's disposition explicitly (green pinning test + zero corpus regressions) for completeness.

File: crates/sim-harness/src/scenario.rs [hunk 3]
```diff
@@ -500,9 +512,15 @@ fn feed_call(out: &mut String, idx: usize, slot: i32, rec: &BoundaryRecord) {
...
+        // slot so adjacent ports' surface caps land ≥2 tiles apart.
+        // Cycle of 4 keeps the ug span (dist-1 ≤ 7) inside the game's
+        // 9-tile limit; same-length repeats are then ≥4 slots apart,
+        // with at least 3 ports physically between them.
+        let dist = 2 + 2 * (slot % 4);
         let _ = writeln!(
             out,
-            "    add_fluid_feed(s, force, head_x, head_y, {ox}, {oy}, \"{item}\")",
+            "    add_fluid_feed(s, force, head_x, head_y, {ox}, {oy}, {dist}, \"{item}\")",
             ox = outward.0,
```
[the remark this answers is not in the document]
Comment (storkme) [reply crates/sim-harness/src/scenario.rs:542]: Accepted — real gap, and you're right that my "the Lua overlap audit backstops this" claim was false for pipes (it only sees chest-on-chest). Reworked in 4e8304ad along your suggested line: ONE ladder per direction with fluids at the FRONT — every fluid tile (out-tiles 1..=2+dist, dist=2+2*slot) sits strictly below the shallowest item band [2+6f, 6+6f]; items shift up by the fluid count, preserving the #363 ordering invariant among themselves. `feed_footprint`'s disjointness guard now models fluid columns too (`fluid_feed_footprint`, conservative whole-column claim), and a 6th same-side fluid fails loudly at codegen (ug 9-tile span) instead of fabricating. New tests: `mixed_fluid_and_item_feeds_share_one_ladder_disjointly` (interleaved columns at 1-tile pitch) + `six_fluid_feeds_on_one_side_panic`. The K60-3 pu3 runs used the interim kit, but their manifest had fluids 28+ tiles from the nearest item rig (verified in the sim dump), so those measurements stand — noted in the RFC decision log.

File: crates/core/src/bus/decomposition_search.rs [hunk 4]
```diff
@@ -1340,17 +1509,21 @@ pub fn select_best_decomposition(
...
+            outcome.as_ref().map(|(_, score)| (i, warn_key, score.score))
         })
-        .max_by(|(ia, a), (ib, b)| {
-            a.partial_cmp(b)
-                .unwrap_or(std::cmp::Ordering::Equal)
-                .then(ib.cmp(ia))
+        // Warnings first (asc), then score (desc), then earliest index —
+        // within the error-free tier a quieter layout beats a denser
+        // one (see the clean_flags comment; RFC-060 decision log).
+        .min_by(|(ia, wa, sa), (ib, wb, sb)| {
```
[the remark this answers is not in the document]
Comment (storkme) [reply crates/core/src/bus/decomposition_search.rs:1533]: Accepted, took your first option — 4e8304ad scopes warnings-first to the native-refusal path (`native_refused = candidates[NATIVE_IDX].0.is_none()`), which is the exact case the reorder was built for (horizontal's 0/6 ec@15 resolution must not outrank a genuinely clean 0/0). The success path keeps #392's original score-then-index order, so every success-path selection is bit-identical to pre-RFC-060 **by construction** rather than by corpus coincidence — which also makes the missing success-path pinning test moot. Full suite green on the scoped comparator; `cell_candidate_resolves_ec15_refusal` continues to pin the refusal side; decision-log entry records the scope correction.
